crates/sargon-uniffi/src/system/interactors/host_interactor.rs

```diff
@@ -0,0 +1,89 @@
+use crate::prelude::*;
+use sargon::KeyDerivationInteractor as InternalKeyDerivationInteractor;
+use sargon::KeyDerivationRequest as InternalKeyDerivationRequest;
+use sargon::KeyDerivationResponse as InternalKeyDerivationResponse;
+use sargon::Result as InternalResult;
+use sargon::SignInteractor as InternalSignInteractor;
+use sargon::Subintent as InternalSubintent;
+use sargon::SubintentHash as InternalSubintentHash;
+use sargon::TransactionIntent as InternalTransactionIntent;
+use sargon::TransactionIntentHash as InternalTransactionIntentHash;
+use sargon::UseFactorSourcesInteractor as InternalUseFactorSourcesInteractor;
+
+type InternalSignRequestForTransactionIntent =
+    sargon::SignRequest<InternalTransactionIntent>;
+type InternalSignWithFactorsOutcomeForTransactionIntent =
+    sargon::SignWithFactorsOutcome<InternalTransactionIntentHash>;
+type InternalSignRequestForSubintent = sargon::SignRequest<InternalSubintent>;
+type InternalSignWithFactorsOutcomeForSubintent =
+    sargon::SignWithFactorsOutcome<InternalSubintentHash>;
+
+/// Sargon os
+#[uniffi::export(with_foreign)]
+#[async_trait::async_trait]
+pub trait HostInteractor: Send + Sync + std::fmt::Debug {
```

Comment:
rename `HostInteractor` -> `UseFactorSourceInteractor`.

Reply:
I disagree a bit with this. There can be a single interactor for hosts. A single place were we can initiate any flow from sargon os.

Reply:
Yes, but this trait is supposed to be a bridge in uniffi. And it is not s trait used in src. From the hosts perspective it is an interactor that interacts with sargon os.

Reply:
> but what about when Sargon needs to tell Host to show some other UI which is not "usage of factor sources".

Can still be defined as a method in `uniffi:HostInteractor` which will translate to a separate trait in src.

Reply:
but why? Why do you wanna go with another system than we have for `Drivers`? For drivers we have many different drivers, they result in different `interface` (kotlin) and `protocol` (swift` in hosts. As they should. Why do you wanna break that (good) pattern with a MotherOfAllTraits which will grow and become a monster interface/protocol?

Reply:
> we cannot present multiple things at the same time. So effectively from the host side we would inject the same component even if we do have multiple interactors.

Exactly from the host's point of view we need a single interactor.

Reply:
I think maybe I get your point, is your point that you WANT to have a **single** Reducer/Activity in hosts which then **dispatches** the requests to child Reducers/sub-Activities?

Reply:
No not at all, I want a single class that implements this interactor to emit events to the application.

The application can subscribe to those events and launch the appropriate screens. Nothing to do with a single Activity.